### PR TITLE
[RFC] ocamlformat: do not use max-indent

### PR DESCRIPTION
**Pull request by nojb**
I propose not to use `max-indent` option of `ocamlformat` which produces misaligned formatting in a number of cases (see the diff). The price to pay is slightly deeper indentation in some cases (see the diff). Personally I think the tradeoff is well worth it.

### CODE REVIEW

**Comment by nojb** [comment]
See https://github.com/ocaml/dune/pull/2580/files#diff-5529262644eb670813336beb1ea001feL102 for sample misalignment.

File: src/dune/expander.ml [hunk 1]
```diff
@@ -372,46 +372,47 @@ let expand_and_record_deps acc ~(dir : Path.Build.t) ~dep_kind
     in
     expand_var t pform syntax_version
     |> Option.bind ~f:(function
-      | Ok s -> Some s
-        | Error (expansion : Pform.Expansion.t) -> (
```
Comment (nojb) [inline src/dune/expander.ml]: Misaligned

File: bin/common.ml [hunk 2]
```diff
@@ -120,11 +120,11 @@ type config_file =
 let default_build_dir = "_build"
 
 (* Allow options from term1 or exclusively options from term2. If the user
-  passes options from both terms, an error is reported. *)
+   passes options from both terms, an error is reported. *)
 let one_of term1 term2 =
   Term.ret
   @@ let+ x, args1 = Term.with_used_args term1
-    and+ y, args2 = Term.with_used_args term2 in
```
Comment (nojb) [inline bin/common.ml]: Misaligned

File: bin/common.mli [hunk 3]
```diff
@@ -21,18 +21,18 @@ val default_target : t -> Arg.Dep.t
...
 
 (** [set_common common ~targets] is [set_dirs common] followed by
-  [set_common_other common ~targets]. In general, [set_common] executes
+    [set_common_other common ~targets]. In general, [set_common] executes
     sequence of side-effecting actions to initialize Dune's working environment
     based on the options determined in a [Common.t] record *)
 val set_common : t -> targets:Arg.Dep.t list -> unit
 
 (** [set_common_other common ~targets] sets all stateful values dictated by
-  [common], except those accounted for by [set_dirs]. [targets] are used to
```
Comment (nojb) [inline bin/common.mli]: Misaligned comments

File: bin/exec.ml [hunk 4]
```diff
@@ -66,8 +66,8 @@ let term =
       |> List.map ~f:(fun p -> Target.Path p)
       |> Target.resolve_targets_mixed common setup
       |> List.concat_map ~f:(function
-        | Ok targets -> targets
-          | Error _ -> []) )
```
Comment (nojb) [inline bin/exec.ml]: Misaligned

Comment (rgrinberg) [reply bin/exec.ml]: This is how `$ make fmt` chooses to do it :/ Let me make sure I have the most recent ocamlformat

**Comment by avsm** [comment]
This is already tracked in ocamlformat as https://github.com/ocaml-ppx/ocamlformat/issues/925

**Comment by nojb** [comment]
Planning to merge this after CI passes.

**Comment by gpetiot** [comment]
I did some tests on your codebase with https://github.com/ocaml-ppx/ocamlformat/pull/997 and it looks better (not perfect though) so far, don't hesitate if you have additional feedback on that.

**Pull request merged**